### PR TITLE
Add install option for cmake

### PR DESCRIPTION
**Pull request by rainhard**
copy include & libs to install directory

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -390,3 +390,16 @@ if (PHOTON_BUILD_TESTING)
...
+################################################################################
+
+# Install headers and libs
+add_custom_target(copy_includes ALL
+    COMMAND ${CMAKE_COMMAND} -E copy_directory ${CMAKE_SOURCE_DIR}/include ${CMAKE_BINARY_DIR}/include
+    COMMENT "Copying include files to build directory: ${CMAKE_BINARY_DIR}/include"
+)
+add_dependencies(copy_includes photon_static)
+
+install(DIRECTORY ${CMAKE_BINARY_DIR}/include/ DESTINATION include)
```
Comment (beef9999) [inline CMakeLists.txt:403]:
```suggestion
install(DIRECTORY ${PROJECT_SOURCE_DIR}/include/ DESTINATION include)
install(FILES ${CMAKE_ARCHIVE_OUTPUT_DIRECTORY}/libphoton.a ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/libphoton.so DESTINATION lib)
```

Comment (beef9999) [reply CMakeLists.txt:403]: Actually we don't need to copy include dir to build dir? So the `copy_includes` target is not necessary. 

And those depentent static libs have been `--whole-archive`ed into libphoton.so, so you can just specify the so file to install.

**Comment by beef9999** [comment]
Will be added in https://github.com/alibaba/PhotonLibOS/pull/628

**Comment by rainhard** [comment]
add `copy_includes` target to copy real files, install DIRECTORY don't follow symlink

File: CMakeLists.txt [hunk 2]
```diff
@@ -390,3 +390,16 @@ if (PHOTON_BUILD_TESTING)
...
+install(TARGETS photon_shared DESTINATION lib)
+install(FILES ${CMAKE_ARCHIVE_OUTPUT_DIRECTORY}/libphoton.a DESTINATION lib)
```
Comment (beef9999) [inline CMakeLists.txt:405]:
```suggestion
install(FILES ${CMAKE_ARCHIVE_OUTPUT_DIRECTORY}/libphoton.a DESTINATION lib)
install(PROGRAMS ${CMAKE_ARCHIVE_OUTPUT_DIRECTORY}/libphoton.so DESTINATION lib)
```

File: CMakeLists.txt [hunk 3]
```diff
@@ -390,3 +390,16 @@ if (PHOTON_BUILD_TESTING)
     endif ()
     GenerateStandaloneCTestScript(${CMAKE_SOURCE_DIR} ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/CTestTestfile.cmake)
 endif ()
+
+################################################################################
+
+# Install headers and libs
+add_custom_target(copy_includes ALL
```
Comment (beef9999) [inline CMakeLists.txt:397]:
```suggestion
add_custom_target(copy_includes ALL
    COMMAND ${CMAKE_COMMAND} -E copy_directory ${PROJECT_SOURCE_DIR}/include ${PROJECT_BINARY_DIR}/include
    COMMENT "Copying include files to build directory: ${PROJECT_BINARY_DIR}/include"
)
add_dependencies(copy_includes photon_static)

install(DIRECTORY ${PROJECT_BINARY_DIR}/include/ DESTINATION include)
```

File: CMakeLists.txt [hunk 4]
```diff
@@ -390,3 +390,22 @@ if (PHOTON_BUILD_TESTING)
...
+    COMMAND ${CMAKE_COMMAND} -E copy_directory ${PROJECT_SOURCE_DIR}/include ${PROJECT_BINARY_DIR}/include
+    COMMENT "Copying include files to build directory: ${PROJECT_BINARY_DIR}/include"
+)
+add_dependencies(copy_includes photon_static)
+
+install(DIRECTORY ${PROJECT_BINARY_DIR}/include/ DESTINATION include)
```
Comment (Coldwings) [inline CMakeLists.txt:403]: is the last bracket a typo?

**Comment by beef9999** [comment]
@rainhard github的这个在Pull Request上直接提commit的功能好像有bug，我提了个包含多行的suggestion，最后出来的不对了...

麻烦重新提一下你最初的PR吧，谢谢。

可以使用`PROJECT_BINARY_DIR` 和 `PROJECT_SOURCE_DIR`，而不是`CMAKE_BINARY_DIR`和`CMAKE_SOURCE_DIR`，因为photon可能被作为submodule子项目，用前者可以保证文件不会出现在最上层项目的build目录

**Review by beef9999 — APPROVED** [review]

**Review by beef9999 — APPROVED** [review]
LGTM

**Pull request merged**